src/wymeditor/editor/native-edit-registration.js

```diff
@@ -19,20 +19,41 @@ WYMeditor.NativeEditRegistration = function (wym) {
     // https:/github.com/mightyiam/edited
     nativeEditRegistration.edited = new WYMeditor.EXTERNAL_MODULES.Edited(
         wym._doc.body,
-        function () {
-            nativeEditRegistration._onNativeEdit.call(nativeEditRegistration);
-        }
+        nativeEditRegistration._onSensibleNativeEdit
+            .bind(nativeEditRegistration),
+        nativeEditRegistration._onAnyNativeEdit
+            .bind(nativeEditRegistration)
     );
 };
 
 /**
-    WYMeditor.NativeEditRegistration._onNativeEdit
-    ==============================================
+    WYMeditor.NativeEditRegistration._onSensibleNativeEdit
+    ======================================================
 
-    Handles native edits for the purpose of Undo/Redo.
+    Handles "sensible" native edits.
+    Sensible according to https://github.com/PolicyStat/edited
 */
-WYMeditor.NativeEditRegistration.prototype._onNativeEdit = function () {
-    var nativeEditRegistration = this;
+WYMeditor.NativeEditRegistration.prototype
+    ._onSensibleNativeEdit = function () {
 
+    var nativeEditRegistration = this;
     nativeEditRegistration.wym.registerModification(true);
 };
+
+/**
+    WYMeditor.NativeEditRegistration._onAnyNativeEdit
+    =================================================
+
+    Handles all native edits.
+*/
+WYMeditor.NativeEditRegistration.prototype._onAnyNativeEdit = function () {
+    var nativeEditRegistration = this,
+        undoRedo = nativeEditRegistration.wym.undoRedo;
+
+    // remove redo points
+    undoRedo.history.changesetsFore = [];
+    // this is a flag for future actions. For example, on undo, when this flag
+    // is true, the state is saved as an undo point before undoing. Otherwise,
+    // this state would have been lost forever.
```

Comment:
No. This explains what it does, not why "this state would have been lost forever". The reason this code needs to exists is because of the difference between sensible edits and any edits, which is an abstraction that can't be assumed for the person reading this code (future us).

The "on undo, when this flag is true, the state is saved as an undo point before undoing." is a bad comment because it's not near the code it's referring to and it's just an english description of some code's behavior (a "what" comment). So it doesn't actually help you understand the code and is likely to lie in the future.